contracts/AelinUpFrontDeal.sol

```diff
@@ -400,41 +410,47 @@ contract AelinUpFrontDeal is AelinERC20, MinimalProxyFactory, IAelinUpFrontDeal
         }
     }
 
+    function claimUnderlyingMutlipleEntries(uint256[] memory _indices) external {
+        for (uint256 i = 0; i < _indices.length; i++) {
+            _claimUnderlying(msg.sender, _indices[i]);
+        }
+    }
+
     /**
-     * @dev purchaser calls after the purchasing period to claim underlying deal tokens
+     * @dev ERC721 deal token holder calls after the purchasing period to claim underlying deal tokens
      * amount based on the vesting schedule
      */
-    function claimUnderlying() external lock purchasingOver passMinimumRaise {
-        uint256 underlyingDealTokensClaimed = claimableUnderlyingTokens(msg.sender);
-        require(underlyingDealTokensClaimed > 0, "no underlying ready to claim");
+    function claimUnderlying(uint256 _tokenId) external {
+        _claimUnderlying(msg.sender, _tokenId);
+    }
+
+    function _claimUnderlying(address _owner, uint256 _tokenId) internal purchasingOver passMinimumRaise {
+        require(ownerOf(_tokenId) == _owner, "must be owner to claim");
```

Comment:
this only requires the passed in address to be the owner. you can claim for other people with this logic. instead the _owner should not be passed. we should enforce the _owner as msg.sender instead. For tax reasons you prob dont want others able to claim for you. maybe we can add a delegate wallet for claiming though. idk

Reply:
It's safe because this function is `internal`. The public one uses `msg.sender`:

```
    function claimUnderlying(uint256 _tokenId) external {
        _claimUnderlying(msg.sender, _tokenId);
    }
  ```

Reply:
And I use the `internal` one in `claimUnderlyingMutlipleEntries` with `msg.sender`